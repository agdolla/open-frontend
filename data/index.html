<!DOCTYPE html><html lang="en"><head><title data-react-helmet="true">Postmates Open Source - Data</title><meta data-react-helmet="true" name="description" content="Open source software at Postmates."/><meta data-react-helmet="true" property="og:title" content="Postmates Open Source - Data"/><meta data-react-helmet="true" property="og:description" content="Open source software at Postmates."/><meta data-react-helmet="true" property="og:image" content="https://engineering.postmates.com/open-frontend/dev-logo.png"/><meta data-react-helmet="true" property="og:type" content="website"/><meta data-react-helmet="true" property="og:url" content="https://engineering.postmates.com/open-frontend/data"/><meta data-react-helmet="true" name="twitter:title" content="Postmates Open Source - Data"/><meta data-react-helmet="true" name="twitter:description" content="Open source software at Postmates."/><meta data-react-helmet="true" name="twitter:creator" content="@Postmates"/><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"/><meta data-react-helmet="true" name="twitter:image" content="https://engineering.postmates.com/open-frontend/dev-logo.png"/><meta data-react-helmet="true" name="twitter:site" content="@Postmates"/><script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-27673166-11"></script><script>
              window.dataLayer = window.dataLayer || [];
              function gtag(){dataLayer.push(arguments);}
              gtag('js', new Date());
              gtag('config', 'UA-27673166-11');
            </script><link rel="preload" as="script" href="https://engineering.postmates.com/templates/vendors~src-pages-ProjectPage.f0e07a7b.js"/><link rel="preload" as="script" href="https://engineering.postmates.com/templates/src-pages-ProjectPage.b83a73c3.js"/><link rel="preload" as="script" href="https://engineering.postmates.com/templates/vendors~main.5cdee4ca.js"/><link rel="preload" as="script" href="https://engineering.postmates.com/main.5452a526.js"/><meta charSet="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="manifest" href="https://engineering.postmates.com/open-frontend/manifest.json"/><link rel="shortcut icon" href="https://engineering.postmates.com/open-frontend/favicon.ico"/></head><body><div id="root"><div style="outline:none" tabindex="-1" role="group"><div><div class="css-n915uf ewpykn40"><div class="css-350yw2 e15cr0n70"><div class="css-hhm5no e15cr0n71"><div class="css-wjeyu7 e15cr0n74"><img src="https://engineering.postmates.com/images/logo.svg" alt="postmates"/><h1 class="css-18a7sf0 e15cr0n72">Postmates <br/> Open Source</h1><p class="css-n09qwt e15cr0n73">We love open-source software. Check out some of the projects we&#x27;ve shared with the community.</p></div><div class="css-1pgq2co e15cr0n75"><div class="css-aynans e1esnmzd0"><div class="css-1sflumz e1esnmzd1"><div class="css-1cbdxus e1esnmzd2"></div><div class="css-tlc6r8 e1esnmzd3"></div><div class="css-vjx4io e1esnmzd6"></div><div class="css-a4fgnr e1esnmzd7"></div><div class="css-121hz9v e1esnmzd5"></div><div class="css-2obzue e1esnmzd4"></div></div></div><div class="css-ake0v9 e1esnmzd0"><div class="css-ne7i8t e1esnmzd1"><div class="css-1cbdxus e1esnmzd2"></div><div class="css-tlc6r8 e1esnmzd3"></div><div class="css-vjx4io e1esnmzd6"></div><div class="css-a4fgnr e1esnmzd7"></div><div class="css-121hz9v e1esnmzd5"></div><div class="css-2obzue e1esnmzd4"></div></div></div></div></div></div><nav class="css-1p6cpjq e1dfarem0"><ul class="css-4c6ebq e1dfarem1"><li class="css-d4wpzs e1dfarem2"><a class="css-1j2fkjx e1dfarem3" href="https://engineering.postmates.com/">All Projects</a></li><li class="css-d4wpzs e1dfarem2"><a class="css-1j2fkjx e1dfarem3" href="https://engineering.postmates.com/frontend">Frontend</a></li><li class="css-d4wpzs e1dfarem2"><a class="css-1j2fkjx e1dfarem3" href="https://engineering.postmates.com/infrastructure">Infrastructure</a></li><li class="css-d4wpzs e1dfarem2"><a class="css-1j2fkjx e1dfarem3" href="https://engineering.postmates.com/android">Android</a></li><li class="css-d4wpzs e1dfarem2"><a class="css-1j2fkjx e1dfarem3" href="https://engineering.postmates.com/ios">iOS</a></li><li class="css-d4wpzs e1dfarem2"><a aria-current="page" class="css-1nzt7dz e1dfarem3" href="https://engineering.postmates.com/data">Data</a></li></ul></nav><div class="css-vooagt e1027l5g0"><main class="css-73w2yp e1027l5g1"></main></div></div></div></div></div><script type="text/javascript">
    window.__routeInfo = {"template":"../src/pages/ProjectPage","sharedHashesByProp":{},"data":{"projects":[],"title":"Postmates Open Source - Data","description":"Open source software at Postmates.","url":"https://engineering.postmates.com/open-frontend/data","siteRoot":"https://engineering.postmates.com/open-frontend"},"path":"data","siteData":{}};</script><script defer="" type="text/javascript" src="https://engineering.postmates.com/templates/vendors~src-pages-ProjectPage.f0e07a7b.js"></script><script defer="" type="text/javascript" src="https://engineering.postmates.com/templates/src-pages-ProjectPage.b83a73c3.js"></script><script defer="" type="text/javascript" src="https://engineering.postmates.com/templates/vendors~main.5cdee4ca.js"></script><script defer="" type="text/javascript" src="https://engineering.postmates.com/main.5452a526.js"></script></body></html>