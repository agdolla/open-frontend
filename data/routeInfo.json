{"template":"../src/pages/ProjectPage","sharedHashesByProp":{},"data":{"projects":[],"title":"Postmates Open Source - Data","description":"Open source software at Postmates.","url":"https://engineering.postmates.com/open-frontend/data","siteRoot":"https://engineering.postmates.com/open-frontend"},"path":"data"}
