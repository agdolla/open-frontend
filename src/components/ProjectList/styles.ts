import styled from 'react-emotion';

export const List = styled.main`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 20px;
  margin-top: 40px;
`;
