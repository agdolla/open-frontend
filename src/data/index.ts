import { Project } from 'types';

export const teams: string[] = [
  'Infrastructure',
  'Frontend',
  'Android',
  'iOS',
  'Data',
];

export const projects: Project[] = [
  {
    title: 'Raster',
    description: 'Testint testing 123',
    language: 'Go',
    team: 'Infrastructure',
    img: '',
    isFeatured : true,
  },
];
