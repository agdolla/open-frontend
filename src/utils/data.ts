import { Project, Language, Teams } from './types';

export const languages: Record<string, Language> = {
  ruby: {
    name: 'Ruby',
    color: '#E72619'
  },
  go: {
    name: 'Go',
    color: '#001DFF'
  },
  mySql: {
    name: 'MySQL',
    color: '#038FEF'
  },
  java: {
    name: 'Java',
    color: '#814C21'
  },
  swift: {
    name: 'Swift',
    color: '#FB8B39'
  },
  javaScript: {
    name: 'JavaScript',
    color: '#F2DD57'
  },
}

export const projects: Project[] = [
  {
    title: 'Raster',
    description: 'Have you ever wondered what your images could look like if it were put through a virtual blender along with a heavy dose of rainbow glitter? Well now you won’t have to wonder—introducing Raster from Postmates.',
    language: languages.go,
    team: Teams.Infrastructure,
    img: '',
    github: '',
    isFeatured: true,
  },
];
