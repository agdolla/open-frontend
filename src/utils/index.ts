import { isArray, isObject, mapValues } from 'lodash';

type func = () => any;

export const is = {
  undef: (v: any): v is undefined => v === null || v === undefined,
  notUndef: (v: any) => v !== null && v !== undefined,
  func: (f: any): f is func => typeof f === 'function',
  number: (n: any): n is number => typeof n === 'number',
  string: (s: any): s is string => typeof s === 'string',
  array: Array.isArray,
  object: (obj: any): obj is object => obj && !is.array(obj) && typeof obj === 'object',
  promise: (p: any): p is Promise<any> => p && is.func(p.then),
  instanceOf: (c: any, p: any) => c && p && c instanceof p,
};

export const once = (() => (
  (cb) => {
    let count = 0;
    return (...args) => {
      if (count === 0) {
        count += 1;
        return cb(...args);
      }
    };
  }
))();

/*
 * Clones any object and recursively replaces null with undefined
 */
export const mapNullToUndefined = (o: any) => {
  if (!isObject(o)) {
    return o === null ? undefined : o;
  }

  return mapValues(o, (v) => {
    if (isArray(v)) {
      return (v as any[]).map(av => mapNullToUndefined(av));
    } else if (isObject(v)) {
      return mapNullToUndefined(v);
    }
    return v === null ? undefined : v;
  });
};

export const noop = (..._: any[]): void => {};
