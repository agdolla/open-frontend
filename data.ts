import { Project, Language, Teams } from './src/utils/types';

export const languages: Record<string, Language> = {
  go: {
    name: 'Go',
    color: '#001DFF',
  },
  mySql: {
    name: 'MySQL',
    color: '#038FEF',
  },
  java: {
    name: 'Java',
    color: '#A5C845',
  },
  kotlin: {
    name: 'Kotlin',
    color: '#EF8E3F',
  },
  swift: {
    name: 'Swift',
    color: '#FB8B39',
  },
  javaScript: {
    name: 'JavaScript',
    color: '#EFD427',
  },
  python: {
    name: 'Python',
    color: '#3A79AC',
  },
  rust: {
    name: 'Rust',
    color: '#814C21',
  },
  elixir: {
    name: 'Elixir',
    color: '#E72619',
  },
};

export const projects: Project[] = [
  {
    title: 'Center Slider View',
    description: 'Recentering slider/seek bar widget for android.',
    language: languages.java,
    team: Teams.Android,
    github: 'https://github.com/postmates/centersliderview-android',
  },
  {
    title: 'Cernan',
    description: 'Telemetry aggregation and shipping, last up the ladder.',
    language: languages.rust,
    team: Teams.Infrastructure,
    github: 'https://github.com/postmates/cernan',
  },
  {
    title: 'Ersa',
    description: 'Utilities for Lightning Components in Visualforce',
    language: languages.javaScript,
    team: Teams.Frontend,
    github: 'https://github.com/postmates/salesforce-ersa',
  },
  {
    title: 'ex-stats-tracker',
    description: 'Elixir statsd client.',
    language: languages.elixir,
    team: Teams.Infrastructure,
    github: 'https://github.com/postmates/ex-stats-tracker',
  },
  {
    title: 'ConsulMutEx',
    description: 'An Elixir module for acquiring and releasing locks with Consul and other backends.',
    language: languages.elixir,
    team: Teams.Infrastructure,
    github: 'https://github.com/postmates/consul_mut_ex',
  },
  {
    title: 'Quantiles',
    description: 'This crate is intended to be a collection of approxiate quantile algorithms that provide guarantees around space and computation.',
    language: languages.rust,
    team: Teams.Infrastructure,
    github: 'https://github.com/postmates/quantiles',
  },
  {
    title: 'Hopper',
    description: 'A Rust mpsc with unbounded members in bounded memory.',
    language: languages.rust,
    team: Teams.Infrastructure,
    github: 'https://github.com/postmates/hopper',
  },
];
