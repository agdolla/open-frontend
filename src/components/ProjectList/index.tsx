import * as React from 'react';
import Project from 'components/Project';
import { Project as ProjectType } from 'utils/types';

import { List } from './styles';

interface Props {
  projects: ProjectType[];
}

const ProjectList = ({ projects }: Props) => (
  <List>
    {projects.map(p => <Project key={p.title} {...p} />)}
  </List>
);

export default ProjectList;
