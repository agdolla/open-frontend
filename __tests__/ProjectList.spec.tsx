import * as React from 'react';
import { shallow } from 'enzyme';
import ProjectList from '../src/components/ProjectList';
import { projects } from '../src/utils/data';

describe('ProjectList', () => {
  it('renders correctly', () => {
    const component = shallow(<ProjectList projects={projects} />);
    expect(component).toMatchSnapshot();
  });
});
