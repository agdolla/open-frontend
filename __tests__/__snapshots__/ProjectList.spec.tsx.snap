// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList renders correctly 1`] = `
<List>
  <Project
    description="Recentering slider/seek bar widget for android."
    github="https://github.com/postmates/centersliderview-android"
    key="Center Slider View"
    language={
      Object {
        "color": "#A5C845",
        "name": "Java",
      }
    }
    team="Android"
    title="Center Slider View"
  />
  <Project
    description="Telemetry aggregation and shipping, last up the ladder."
    github="https://github.com/postmates/cernan"
    key="Cernan"
    language={
      Object {
        "color": "#814C21",
        "name": "Rust",
      }
    }
    team="Infrastructure"
    title="Cernan"
  />
  <Project
    description="Utilities for Lightning Components in Visualforce"
    github="https://github.com/postmates/salesforce-ersa"
    key="Ersa"
    language={
      Object {
        "color": "#EFD427",
        "name": "JavaScript",
      }
    }
    team="Frontend"
    title="Ersa"
  />
  <Project
    description="Elixir statsd client."
    github="https://github.com/postmates/ex-stats-tracker"
    key="ex-stats-tracker"
    language={
      Object {
        "color": "#E72619",
        "name": "Elixir",
      }
    }
    team="Infrastructure"
    title="ex-stats-tracker"
  />
  <Project
    description="An Elixir module for acquiring and releasing locks with Consul and other backends."
    github="https://github.com/postmates/consul_mut_ex"
    key="ConsulMutEx"
    language={
      Object {
        "color": "#E72619",
        "name": "Elixir",
      }
    }
    team="Infrastructure"
    title="ConsulMutEx"
  />
  <Project
    description="This crate is intended to be a collection of approxiate quantile algorithms that provide guarantees around space and computation."
    github="https://github.com/postmates/quantiles"
    key="Quantiles"
    language={
      Object {
        "color": "#814C21",
        "name": "Rust",
      }
    }
    team="Infrastructure"
    title="Quantiles"
  />
  <Project
    description="A Rust mpsc with unbounded members in bounded memory."
    github="https://github.com/postmates/hopper"
    key="Hopper"
    language={
      Object {
        "color": "#814C21",
        "name": "Rust",
      }
    }
    team="Infrastructure"
    title="Hopper"
  />
</List>
`;
